internal/operators/cnv/cnv_operator.go

```diff
@@ -0,0 +1,139 @@
+package cnv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/hardware"
+	"github.com/openshift/assisted-service/internal/operators/api"
+	models "github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+type operator struct {
+	log logrus.FieldLogger
+}
+
+// NewCNVOperator creates new instance of a Container Native Virtualization installation plugin
+func NewCNVOperator(log logrus.FieldLogger) *operator {
+	return &operator{
+		log: log,
+	}
+}
+
+// GetType reports type of an operator this Operator manages
+func (o *operator) GetType() models.OperatorType {
+	return models.OperatorTypeCnv
+}
+
+// GetDependencies provides a list of dependencies of the Operator
+func (o *operator) GetDependencies() []models.OperatorType {
+	return []models.OperatorType{models.OperatorTypeLso}
+}
+
+// GetClusterValidationID returns cluster validation ID for the Operator
+func (o *operator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDCnvRequirementsSatisfied)
+}
+
+// GetHostValidationID returns host validation ID for the Operator
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDCnvRequirementsSatisfied)
+}
+
+// ValidateCluster always return "valid" result
+func (o *operator) ValidateCluster(ctx context.Context, cluster *common.Cluster) (api.ValidationResult, error) {
+	var insufficientHosts []string
+	var status string
+	for _, host := range cluster.Hosts {
```

Comment:
This loop seems to be superfluous; `ValidateHost` is called byt the `internal/host` package validation: https://github.com/openshift/assisted-service/blob/10bd870d5a010c119b3293d2c0f9294760ade8ff/internal/host/refresh_status_preprocessor.go#L53

Reply:
It seems like good idea to share the logic. How do you suggest to change it?

Reply:
But do you need to share it if it's validated somewhere else?

Reply:
I wanted to implement cluster level validation by reusing what we have in `ValidateHost`. As far as I understand there is nothing else to validate for a cluster.

Reply:
Maybe the `ValidateCluster` method shouldn't do any validation and leave it to the `ValidateHost`?
@gamli75 - what do you think?

Reply:
if all the CNV validations are per-host - we can use only validateHost method, there is no need to display the same validation error both on the host & cluster. In OCS there is a meaning for validateCluster since they aggregate all hosts' CPUs.